src/Log/Migrations/CreateNewLogTables.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Give\Log\Migrations;
+
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Framework\Database\Exceptions\DatabaseQueryException;
+use Give\Framework\Migrations\Exceptions\DatabaseMigrationException;
+
+
+
+/**
+ * Class CreateNewLogTables
+ * @package Give\Log\Migrations
+ *
+ * @since 2.9.7
+ */
+class CreateNewLogTables extends Migration {
+
+	/**
+	 * @return string
+	 */
+	public static function id() {
+		return 'create_new_log_table';
+	}
+
+	/**
+	 * @return int
+	 */
+	public static function timestamp() {
+		return strtotime( '2020-01-28 12:00' );
+	}
+
+
+	public function run() {
+		global $wpdb;
+
+		$sql = "CREATE TABLE {$wpdb->prefix}give_log (
+  			id BIGINT UNSIGNED NOT NULL AUTO_INCREMENT,
+  			log_type VARCHAR(16) NOT NULL,
+  			migration_id VARCHAR(64) NULL,
+  			data text NOT NULL,
+  			category VARCHAR(64) NULL,
+  			source VARCHAR(64) NULL,
+  			date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  			PRIMARY KEY  (id),
+            KEY log_type (log_type),
+            KEY migration_id (migration_id),
+            KEY category (category),
+            KEY source (source)
+		) {$wpdb->get_charset_collate()}";
+
+		try {
+			DB::delta( $sql );
+		} catch ( DatabaseQueryException $exception ) {
+			throw new DatabaseMigrationException( 'An error occurred creating the give_log table: ' . print_r( $exception->getQueryErrors(), true ) );
```

Comment:
I see what you're doing here. Do yourself (and others) a favor and make this possible:
```php
throw DatabaseMigrationException::fromException($exception, 'An error occurred...');
```

Reply:
@JasonTheAdams I'm not sure is this what you meant and did I implement it right. 

71087fb2